dds_cli/user.py

```diff
@@ -312,40 +309,30 @@ def token_report(self, token_metadata=None):
         LOG.info(f"[{markup_color}]{sign}  {message} {sign} [/{markup_color}]")
         LOG.info(f"[{markup_color}]Token age: {readable_timedelta(age)}[/{markup_color}]")
 
-        if lifetime.total_seconds() < 0:
+        if age > dds_cli.TOKEN_MAX_AGE:
+            LOG.info(f"[{markup_color}]Token expired: {expiration_time}[/{markup_color}]")
+        else:
+            lifetime = age - dds_cli.TOKEN_MAX_AGE
             LOG.info(
                 f"[{markup_color}]Token expires: {expiration_time} (in {readable_timedelta(lifetime)})[/{markup_color}]"
             )
-        else:
-            LOG.info(f"[{markup_color}]Token expired: {expiration_time}[/{markup_color}]")
-
-        if consignee:
-            LOG.info(f"[{markup_color}]Token issued to: {consignee}[/{markup_color}]")
 
     # Private methods ############################################################ Private methods #
-    def __token_dates(self, token_metadata):
-        """Returns definitive or estimated values for the issue date, token's age, lifetime and expiration time in UTC."""
-
-        local_tz = tzlocal.get_localzone()
-        utc_tz = pytz.timezone("UTC")
-
-        # Try to use the Issued At Claim (iat), otherwise fall back to modification_time of the file
-        if token_metadata and token_metadata.get("iat", None):
-            issued_at = datetime.datetime.fromtimestamp(token_metadata.get("iat"), tz=utc_tz)
-        else:
-            issued_at = datetime.datetime.fromtimestamp(
-                os.path.getmtime(self.token_file), tz=local_tz
-            )
-
-        age = datetime.datetime.utcnow().replace(tzinfo=utc_tz) - issued_at
-
-        # Try to use the Expiration Time (exp), otherwise fall back to calculation based on configured lifetime value
-        if token_metadata and token_metadata.get("exp", None):
-            expiration_time = datetime.datetime.fromtimestamp(token_metadata.get("exp"), tz=utc_tz)
+    def __token_dates(self, token):
+        """Returns definitive (based on the token jose header) or estimated values
+        (based on the token file) for the token's age and expiration time in UTC."""
+
+        expiration_time = get_token_expiration_time(token=token)
+
+        # Try to use the Expiration Time (exp) in the token jose header,
+        # otherwise fall back to calculation based on the token file
+        if expiration_time:
+            utc_tz = pytz.timezone("UTC")
+            expiration_time = datetime.datetime.fromtimestamp(expiration_time, tz=utc_tz)
+            age = dds_cli.TOKEN_MAX_AGE - (expiration_time - datetime.datetime.utcnow())
```

Comment:
If you do not want the time when the token was issued to the token itself for length constraints, then why don't you use the `modification time` here, too? 

I think this is a better approximation than a configured the `dds_cli.TOKEN_MAX_AGE` configured value, which I would prefer to only have as a fallback if possible (but I can also be convinced otherwise - more of a gut feeling)

Reply:
modification time depends on token file, where you comment is a token header. They are unrelated. If exp in token header, we use that. Issue time is not a required claim and we don't need it at least for now. Users need to deal with the expiration time, because after that the token will be expired. Knowing the issue time of a token probably does not help a user.

Reply:
I agree. I kept the `TOKEN_MAX_AGE` and `TOKEN_WARNING_AGE` because those were initially introduced by Johannes. Shouldn't we then not replace both with a `REMAINING_LIFETIME_WARNING` timedelta? Such that we don't bother at all how old the token is, but just warn once there are only X hours of lifetime left until it expires?

Reply:
I think a notification that warns that the token will soon expire is good enough. For example, this is how the GitHub token works. I generate a token and forget about it until GitHub tells me that there are a few days left until it expires, so I go and regenerate it. We don't need to bother with age and etc. The expiration time gives us the deadline.

Reply:
Yes, sounds good. And the warning limit for soon could be a locally configured value like `REMAINING_LIFETIME_WARNING_PERIOD` or such.

Reply:
I agree, warning when x hours remain of the token is good.